exp/runtime/internal/controllers/extension_controller.go

```diff
@@ -197,14 +200,59 @@ func (r Reconciler) discoverExtension(ctx context.Context, extension *runtimev1.
 // if any of these checks fails the response is invalid and an error is returned. Extensions with previously valid
 // RuntimeExtension registrations are not removed from the registry or the object's status.
 func validateRuntimeExtensionDiscovery(ext *runtimev1.Extension) error {
+	names := make(map[string]bool)
+	var errs []error
+	// TODO: Implement the same validation rules in the runtime SDK server side Discovery handler.
 	for _, runtimeExtension := range ext.Status.RuntimeExtensions {
-		// TODO: Extend the validation to cover more cases.
 
-		// simple dummy check to show how and where RuntimeExtension validation works.
-		// this should aggregate the errors on validation.
-		if len(runtimeExtension.Name) > 63 {
-			return errors.New("RuntimeExtension name must be less than 64 characters")
+		// Names should be unique
+		if _, ok := names[runtimeExtension.Name]; ok {
+			errs = append(errs, errors.Errorf("duplicate name %s found", runtimeExtension.Name))
+		}
+
+		names[runtimeExtension.Name] = true
+		// Name should match Kubernetes naming conventions - validated based on Kubernetes DNS1123 Subdomain rules.
+		if errStrings := validation.IsDNS1123Subdomain(runtimeExtension.Name); len(errStrings) > 0 {
+			errs = append(errs, errors.Errorf("Name %s is not valid: %s", runtimeExtension.Name, errStrings))
 		}
+
+		// Timeout should be a positive integer and should be under 600 seconds (10 minutes)
+		if *runtimeExtension.TimeoutSeconds < 0 && *runtimeExtension.TimeoutSeconds > 600 {
```

Comment:
10 minutes for maximum timeout seems too long. The admission webhooks in k8s allow a maximum timeout of 30s. 
How about using the same for our maximum as well?